.github/workflows/publish-to-test-pypi.yml

```diff
@@ -0,0 +1,90 @@
+name: Publish Python 🐍 distribution 📦 to PyPI
+
+on: push
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.10.12"
+    - name: Install specific setuptools and pip
+      run: |
+        python -m pip install setuptools==65.5.0 pip==21
+    - name: Install pypa/build
+      run: |
+        python3 -m pip install build --user
+    - name: Build a binary wheel and a source tarball
+      run: python3 -m build
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
+    name: >-
+      Publish Python 🐍 distribution 📦 to PyPI
+    if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
```

Comment:
So this is triggered by pushing a new tag? Does this mean we still need to push a new tag? 
Should this not be triggered by a push to main instead? And then we can grab the version from src/nett/_version.py. 

This is how I would imagine it:
1. Check if this is a PR to main AND that the version found in src/nett/_version.py is not an existing release (denoted as v{version num}): If both are true, run the build to confirm it is buildable.
2. If this is being pushed to main (from merging the PR): Start the publication process, resulting in it being tagged and published to pypi.

The particular if statement solution for seeing if it is being pushed to main or is just a PR for main can be found in the docs workflow, see https://github.com/buildingamind/NewbornEmbodiedTuringTest/blob/bf29953ab909ae216eb67a06fb9dbda281b9def1/.github/workflows/docs.yml#L3C1-L11C12

Reply:
To be honest I don't have an opinion, I'm fine if we do this either way, but if what you have is already how `docs` work, it would make all the sense to have consistency between the two workflows.